migrations/statictypes/statictype_migration.go

```diff
@@ -246,16 +250,36 @@ func (m *StaticTypeMigration) maybeConvertStaticType(staticType interpreter.Stat
 		}
 
 	case *interpreter.InterfaceStaticType:
+		var convertedType interpreter.StaticType
 		interfaceTypeConverter := m.interfaceTypeConverter
 		if interfaceTypeConverter != nil {
-			return interfaceTypeConverter(staticType)
+			convertedType = interfaceTypeConverter(staticType)
 		}
 
+		// Interface types need to be placed in intersection types
+		if _, ok := parentType.(*interpreter.IntersectionStaticType); !ok {
+			if convertedType == nil {
+				convertedType = interpreter.NewIntersectionStaticType(
+					nil, []*interpreter.InterfaceStaticType{
+						staticType,
+					},
+				)
+			} else if convertedInterfaceType, ok := convertedType.(*interpreter.InterfaceStaticType); ok {
```

Comment:
why does this second if-condition needed? is it to avoid any accidental double-wrapping of  `IntersectionStatic` types? (e.g: type-converter already returns the correctly wrapped type)

Reply:
I added comments for the two cases in 2662be331fc51bf2efff26356fbc0c35fc015f93.

Do they answer your question?

Reply:
yes - thank you for the explanation!

I just realized, do we need to do a similar check for the returned value of `m.compositeTypeConverter(staticType)`? e.g: what if a composite type was converted to an interface type? Think we don't have such conversions yet, but just to be future proof?

Reply:
Good idea! We should maybe add that as well and add a test 👍

Reply:
> Think we don't have such conversions yet

actually this is needed when refactoring type-requirements (composite type would become interfaces). e.g: `@FungibleToken.Vault` -->  `@{FungibleToken.Vault}`

Reply:
nice!

Reply:
@SupunS Added in fec43976e36dcbe632077ed981046821acaa647b. Also added tests in general for this intersection type wrapping for both composite and interface type converters